src/raygun.rum.js

```diff
@@ -291,6 +291,7 @@ var raygunRumFactory = function (window, $, Raygun) {
             if(isPageOrVirtualPage) {
               // If the next timing data is a page or virtual page, generate a new request ID
               createRequestId();
+              self.postAttempts[self.requestId] = 0;
```

Comment:
The requestId is still a global variable. Somehow that needs to be assigned to each request as well, otherwise the same issue as before could end up happening. I think in the example situation of one network request stalling and then another goes out before the first is retried will result in the same problem.

I'm wondering if perhaps a better approach would be not actually send the payloads when they get sent, but put them into a queue and then have some background process (perhaps using setInterval) that takes some number off the queue and attempts to send them all, if they succeed that's the end of it, if they fail they get put back onto the queue (probably at the front)

Reply:
Thanks, I'll look into this